### ISSUE TITLE
Have the active item always be in center when showing multiples

### ISSUE THREAD

**Issue by paperscissors** [antoniandre/vueper-slides#56]
This is great, thank you for your hard work.

I wonder, is it possible to have the active slide always in center, say when showing three slides at the same time? 

I refer to this situation: https://antoniandre.github.io/vueper-slides/#ex--show-multiple-slides

My ideal is to sort of like how OSX and Windows do slideshows of images in their respective file managers, or in some games, how you select a level.

To try to illustrate, here is a rough diagram:

```
* = no slide
x = visible slide, but not active
X = visible slide that is active
```

So, say you have 3 slides. The slider loads like this, with slide one in second position:

`* X x`

Then slides advanced:
`x X x`

Finally if you advance to the last option, you'd see:
`x X *`

Does this make sense, and is this possible (without infinite slides, of course)? 

Just to give an illustration of what I essentially mean, layout wise:
<img width="100%" alt="Screen Shot 2019-11-13 at 3 25 32 PM" src="https://user-images.githubusercontent.com/15576/68813188-ec884200-0629-11ea-9647-a0abe6b1136f.png">

Thanks!

**Comment by paperscissors** [comment]
You know, I actually managed to figure this out with a little extra CSS binding and watching the `@before-slide` event.

It remains to be seen what this will do responsively, but I think it may work since it's all percentage based.

So, on the component:

`<vueper-slides class="no-shadow" :class="{ offsetslidefirst: first_slide, offsetslidelast: last_slide }" @before-slide="beforeSlide" :visible-slides="3" :slide-ratio="1/4" :dragging-distance="35">`

Callback method:

```
beforeSlide (eventName, params) {
      this.first_slide = (params.nextSlide.index === 0)
      this.last_slide = (params.nextSlide.index === this.slides.length - 1)
    }
```

And styles:
```
.offsetslidefirst {
  .vueperslides__track-inner {
    transform: translate3d(33.333%, 0px, 0px) !important;
  }
}

.offsetslidelast {
  .vueperslides__track-inner {
    transform: translate3d(-99%, 0px, 0px) !important;
  }
}
```

There may be a more elegant way to do this, but it's working for me. Thanks!

![proof-of-concept](https://user-images.githubusercontent.com/15576/68818566-3aa54180-063a-11ea-8415-37e579d4f417.gif)

**Comment by antoniandre** [comment]
Hi @paperscissors, this is great, a good way of doing it! 
Thanks for sharing your code, I might reuse it as an example of the documentation later.

**Comment by paperscissors** [comment]
My pleasure, @antoniandre !

**Comment by antoniandre** [comment]
Hi @paperscissors,
Just letting you know that the V2 is released, this is what it will change for this code:

```js
beforeSlide ({ nextSlide }) {
      this.first_slide = (nextSlide.index === 0)
      this.last_slide = (nextSlide.index === this.slides.length - 1)
    }
```

The first parameter eventName is removed as I realised it was useless.
You might have custom arrows too, unless it's a border, arrows slots have been renamed to `arrow-left` and `arrow-right`.

Cheers.

**Comment by paperscissors** [comment]
@antoniandre thanks for the heads up, this makes sense!

**Comment by stylianospanagakos** [comment]
Hi @antoniandre,

The CSS solution that was given works only if the total number of slides equals to 5. For any other dataset where the number of slides is unknown beforehand, the `translate3d` property needs to have the following calculation in order to determine the offset of `vueperslides__track-inner` :

`33% - (currentSlideIndex * 33%)`

Is there a way to expose the `vueperslides__track-inner` element as a child component in order to bind a `style` property to it and use the above calculation?

Thanks for the great package!

**Comment by paperscissors** [comment]
@stylianospanagakos Fair point! I'm actually just realizing this now as I've added some more elements to the display I'm working with. I did just try to upgrade to the newest but realizing that a lot of CSS classes seem to have changed since I came up with this code.

**Comment by paperscissors** [comment]
@stylianospanagakos I figured out an interim way to do what you're proposing. Not the cleanest or most semantic way, but it works!

In the same Vue component:

```
<template>
  <div class="container, or wherever you want" :style="slideOffset">
<vueper-slides ... @before-slide="beforeSlide" :visible-slides="3" :slide-ratio="1/4" :dragging-distance="35">
```

```
  computed: {
    slideOffset () {
      return {
        '--offset': this.currentSlide
      }
    }
  },
```

```
    beforeSlide (slides) {
      let currentSlide = slides.nextSlide !== undefined ? slides.nextSlide.index : 0 // just to cover bases here
      this.first_slide = (currentSlide === 0)
      this.last_slide = (currentSlide === this.games.length - 1)
      this.currentSlide = currentSlide
    },
```

```
<style lang="scss">
.vueperslides__track-inner {
    transform: translate3d(calc(33% - (var(--offset) * 33%)), 0px, 0px) !important;
 }
</style>
```

**Comment by paperscissors** [comment]
You know, one odd thing I am encountering with this is that when the page loads, it doesn't give the first slide or any programmatically set slide the `vueperslide--active vueperslide--visible` classes. It does if I come back to the slide instance with `this.$router.push` or if Vue reloads the page after I make a code change, but it's not on first page load.

I'm trying to manually add the classes in by looping through `this.$refs.gameSelector.$children` and using classList.add without any luck. 

I don't think it has to do with what I've done here since it seems to happen even without interacting with the API. It's rather strange. I think it wasn't doing this when I had an earlier version and then started when I upgraded, but I can't pinpoint exactly when that happened. I am guess though that it was when I bumped up to V2.
